gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/retention/HivePartitionVersionRetentionCleanerPolicy.java

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.compliance.retention;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Preconditions;
+
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.compliance.ComplianceConfigurationKeys;
+import gobblin.compliance.HivePartitionDataset;
+import gobblin.compliance.HivePartitionVersion;
+import gobblin.compliance.HivePartitionVersionPolicy;
+import gobblin.configuration.State;
+
+
+/**
+ * A retention version policy for the {@link HivePartitionRetentionVersion}.
+ *
+ * @author adsharma
+ */
+@Slf4j
+public class HivePartitionVersionRetentionCleanerPolicy extends HivePartitionVersionPolicy {
+  /**
+   * Maximum number of backups to be retained.
+   */
+  private int backupRetentionVersions;
+  /**
+   * Maximum age for a retained backup.
+   */
+  private int backupRetentionDays;
+  /**
+   * Maximum age for a retained trash partition.
+   */
+  private int trashRetentionDays;
+
+  public HivePartitionVersionRetentionCleanerPolicy(State state, HivePartitionDataset dataset) {
+    super(state, dataset);
+    Preconditions.checkArgument(this.state.contains(ComplianceConfigurationKeys.CLEANER_BACKUP_RETENTION_VERSIONS),
+        "Missing required property " + ComplianceConfigurationKeys.CLEANER_BACKUP_RETENTION_VERSIONS);
+    Preconditions.checkArgument(this.state.contains(ComplianceConfigurationKeys.CLEANER_BACKUP_RETENTION_DAYS),
+        "Missing required property " + ComplianceConfigurationKeys.CLEANER_BACKUP_RETENTION_DAYS);
+    Preconditions.checkArgument(this.state.contains(ComplianceConfigurationKeys.CLEANER_TRASH_RETENTION_DAYS),
+        "Missing required property " + ComplianceConfigurationKeys.CLEANER_TRASH_RETENTION_DAYS);
+    this.backupRetentionVersions =
+        this.state.getPropAsInt(ComplianceConfigurationKeys.CLEANER_BACKUP_RETENTION_VERSIONS);
+    this.backupRetentionDays = this.state.getPropAsInt(ComplianceConfigurationKeys.CLEANER_BACKUP_RETENTION_DAYS);
+    this.trashRetentionDays = this.state.getPropAsInt(ComplianceConfigurationKeys.CLEANER_TRASH_RETENTION_DAYS);
+  }
+
+  @Override
+  public boolean shouldSelect(HivePartitionVersion version) {
+    // not implemented by this class
+    return false;
+  }
+
+  @Override
+  public List<HivePartitionVersion> selectedList(List<HivePartitionVersion> versions) {
+    if (versions.isEmpty()) {
+      return versions;
+    }
+    List<HivePartitionRetentionVersion> backupVersions = new ArrayList<>();
+    List<HivePartitionRetentionVersion> trashVersions = new ArrayList<>();
+    List<HivePartitionVersion> selectedVersions = new ArrayList<>();
+    for (HivePartitionVersion version : versions) {
+      String prefix = this.dataset.getDbName();
+      if (!version.getTableName().startsWith(prefix)) {
```

Comment:
Using a prefix of db name + "_" would be better to avoid matching another table that has a matching prefix.